controllers/storagecluster/storageconsumer.go

```diff
@@ -208,6 +208,15 @@ func (s *storageConsumer) ensureDeleted(r *StorageClusterReconciler, storageClus
 			return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from StorageConsumer: %v", err)
 		}
 	}
+
+	consumerConfigMap := &corev1.ConfigMap{}
+	consumerConfigMap.Name = defaults.LocalStorageConsumerConfigMapName
+	consumerConfigMap.Namespace = storageCluster.Namespace
+
+	if err := r.Delete(r.ctx, consumerConfigMap); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to delete local consumerConfigMap %s: %v", consumerConfigMap.Name, err)
+	}
```

Comment:
why should this be removed explicitly? either deletion of storagecluster or storageconsumer removes this, isn't it?

Reply:
I believe the deletion will automatically get triggered by the consumer deletion, We need to remove the finalizer from the configmap which blocks it's deletion.

Reply:
you response has a lot of info and some of it needs to be added as a comment, specifically the cyclic relationship which is the main reason for us not waiting for storagecluster deletion.

btw, don't we need any check that this CM is not being owned by 2 consumers? better safe than sorry even though MDR is for external client only?

Reply:
iirc there is some effort around reinstating unintentional storagecluster deletion and it could help if we can have this CM around, that means not deleting this at all.

will let other reviewers or Ohad to decide.